src/modules/move/MoveModal.jsx

```diff
@@ -96,57 +89,83 @@ const MoveModal = ({ onClose, entries }) => {
       return
     }
 
-    moveEntries(folderId)
+    moveEntries(folder)
   }
 
-  const moveEntries = async folderId => {
+  const moveEntries = async folder => {
     try {
       setMoveInProgress(true)
       const trashedFiles = []
       await Promise.all(
         entries.map(async entry => {
-          const targetPath = await registerCancelable(
-            CozyFile.getFullpath(folderId, entry.name)
-          )
-          const force = !sharedPaths.includes(targetPath)
+          const force = !sharedPaths.includes(folder.path)
           const moveResponse = await registerCancelable(
-            CozyFile.move(entry._id, { folderId }, { force })
+            move(client, entry, folder, {
+              force
+            })
           )
           if (moveResponse.deleted) {
             trashedFiles.push(moveResponse.deleted)
           }
         })
       )
 
-      const response = await registerCancelable(
-        client.query(Q('io.cozy.files').getById(folderId))
-      )
-      const targetName = response.data?.name || t('breadcrumb.title_drive')
-      const targetDir = response.data?.id
+      const isMovingInsideNextcloud =
+        folder._type === 'io.cozy.remote.nextcloud.files'
+      if (isMovingInsideNextcloud) {
+        client.resetQuery(
+          computeNextcloudFolderQueryId({
+            sourceAccount: folder.cozyMetadata.sourceAccount,
+            path: folder.path
+          })
+        )
+      }
+
+      const isMovingOutsideNextcloud =
+        !isMovingInsideNextcloud &&
+        entries[0]._type === 'io.cozy.remote.nextcloud.files'
+      if (isMovingOutsideNextcloud) {
+        client.resetQuery(
+          computeNextcloudFolderQueryId({
+            sourceAccount: entries[0].cozyMetadata.sourceAccount,
+            path: getParentPath(entries[0].path)
+          })
+        )
+      }
+
+      // Gérer le cas où le target est un shortcut
+      const targetName = folder.name || t('breadcrumb.title_drive')
+
+      // Comment récupérer l'id du shortcut ici
+      const targetRoute =
+        folder._type === 'io.cozy.remote.nextcloud.files'
+          ? `/nextcloud/${folder.cozyMetadata.sourceAccount}?path=${folder.path}`
+          : `/folder/${folder.id}`
```

Comment:
Refactor the logic for resetting the Nextcloud query and determining the target route into separate helper functions. This improves readability and maintainability by keeping the component focused on rendering
for instance:

```jsx
const handleResetNextcloudQuery = useCallback(() => {
  const isMovingInsideNextcloud =
    folder._type === 'io.cozy.remote.nextcloud.files';
  if (isMovingInsideNextcloud) {
    client.resetQuery(
      computeNextcloudFolderQueryId({
        sourceAccount: folder.cozyMetadata.sourceAccount,
        path: folder.path
      })
    );
  }

  const isMovingOutsideNextcloud =
    !isMovingInsideNextcloud &&
    entries[0]._type === 'io.cozy.remote.nextcloud.files';
  if (isMovingOutsideNextcloud) {
    client.resetQuery(
      computeNextcloudFolderQueryId({
        sourceAccount: entries[0].cozyMetadata.sourceAccount,
        path: getParentPath(entries[0].path)
      })
    );
  }
}, [client, folder, entries]);

const getTargetRoute = useCallback(() => {
  return folder._type === 'io.cozy.remote.nextcloud.files'
    ? `/nextcloud/${folder.cozyMetadata.sourceAccount}?path=${folder.path}`
    : `/folder/${folder.id}`;
}, [folder]);

useEffect(() => {
  handleResetNextcloudQuery();
}, [handleResetNextcloudQuery]);
```

Reply:
I've tried to make it easier to read, so don't hesitate to let me know if you find it too complex